src/components/betting-chips/BettingChip.tsx

```diff
@@ -14,18 +15,34 @@ export type BettingChipProps = {
 
 export const BettingChip = observer((props: BettingChipProps) => {
   const { value, betOn } = props;
-  const { game } = useMainStore();
+  const { game, player, createSnapshot } = useMainStore();
+  const bettingChipSound = new Audio(sound);
 
   const addToBet = () => {
     switch (betOn) {
       case "player":
-        game.addToPlayerBet(value);
+        if (value <= player.playerMoney) {
+          bettingChipSound.play();
+          game.addToPlayerBet(value);
+          player.removePlayerMoney(value);
+          createSnapshot();
```

Comment:
This logic must be inside MobX store. You should try structure your code keeping in mind two rules:
- everything that is related to business logic must be inside store (mobx)
- everything that is related to UI components must be inside of it (audio, component logic)